policies/recipes/artifacts_test.go

```diff
@@ -0,0 +1,302 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package recipes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	osconfigpb "github.com/GoogleCloudPlatform/osconfig/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha2"
+	"github.com/fsouza/fake-gcs-server/fakestorage"
+)
+
+func TestFetchArtifacts_GCS_happyCase(t *testing.T) {
```

Comment:
downloading artifacts is not something that should be unit tested. this test provides no value, as the failure scenarios likely to exist are not going to occur with mocks or fake servers.

Reply:
we can simulate the failure scenrios with the mocks. that is the whole point of having mocks or fake servers. i am talking about failure scenrios when talking to external system like network failures;
the whole point of adding these unit tests are to check if our application handles these failures from external system gracefully or not

Reply:
please don't raise old discussions; we don't want to add mocks here.